### ISSUE TITLE
Sorting the question in the table view will add a second sort in the same field breaking the query

### ISSUE THREAD

**Issue by paoliniluis** [metabase/metabase#46992]
### Describe the bug

This seems to be SQL server specific but I'm not 100% sure (tested on H2 and I'm not able to trigger it)

### To Reproduce

1) set up a Metabase with a SQL server
2) build a question like 
![image](https://github.com/user-attachments/assets/b9713ed3-3976-4daa-ac4f-4a4db2737eb3)
3) then click on visualize and later on sort by the created at field  (use the table view)
![image](https://github.com/user-attachments/assets/aa219566-b919-4f26-9d49-74aa4fbecf6b)

see what just happened (a double sort making the query invalid)
![image](https://github.com/user-attachments/assets/55754a3f-67fd-4d3e-83c0-d9d23b492591)

### Expected behavior

We shouldn't inject a sort twice in the query

### Logs

NA

### Information about your Metabase installation

```JSON
this has been around for ages
```

### Severity

P1

### Additional context

reported by an enterprise customer

**Comment by bshepherdson** [comment]
I was able to reproduce this, but only on SQL Server.

The proximal cause is that while adding the implicit `ORDER BY` clauses for breakouts, we fail to recognize that there's already a sort order for Created At.

The breakout is spelled
```clojure
[:field 123 {:base-type :type/DateTime, :temporal-unit :month}]
```
whereas the `:order-by` is spelled
```clojure
[:desc [:field 123 {:base-type :type/Date, :temporal-unit :month}]]
```

That is, the breakout has the correct `:base-type :type/DateTime` while the sort order was added with a (halfway-correct due to the truncation) `:type/Date`. This matching code uses direct set matching, so it does not allow for any differences.

I'm continuing to investigate for the root cause. It seems that on H2 and Postgres the sort order has `:base-type :type/DateTime` and this issue does not appear.
